### PR TITLE
Rat changes

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

This does:
1. Change some mouse code to not look horrific
2. Adds mouse language to regal rats and normal rats
3. Allows rats to eat food
4. Allows ghosts to posses rats
5. Makes sentient rats not chew cables randomly
6. Makes rats coloured
7. Makes royal cheese spawned rats have names, same as event spawned ones

# Wiki Documentation

Mous

# Changelog

:cl:  
rscadd: Rats and Regal rats can now eat food to heal
rscadd: Rats now speak mouse language and are colored
rscadd: Rats now can be possessed by ghosts
tweak: Makes mouse CtrlClick code look less horrible
tweak: Moves rat code into a new file

/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 1]
```diff
@@ -183,22 +183,23 @@ GLOBAL_VAR_INIT(mouse_killed, 0)
...
 	layer = MOB_LAYER
-	visible_message(span_danger("[src] starts eating away [A]..."),
-						 span_notice("You start eating the [A]..."))
+	visible_message(span_danger("[src] starts eating away [A]..."),span_notice("You start eating the [A]..."))
 	if(do_after(src, 3 SECONDS, FALSE, A))
 		if(QDELETED(A))
 			return
-		visible_message(span_danger("[src] finishes eating up [A]!"),
-						 span_notice("You finish up eating [A]."))
+		visible_message(span_danger("[src] finishes eating up [A]!"),span_notice("You finish up eating [A]."))
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:202]: it looks tidier while in order

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 2]
```diff
@@ -301,26 +218,18 @@
 		if (do_mob(src, target, 2 SECONDS))
 			target.reagents.add_reagent(/datum/reagent/rat_spit,rand(1,3),no_react = TRUE)
 			to_chat(src, span_notice("You finish licking [target]."))
+	else if(istype(target, /obj/item/reagent_containers/food/snacks/cheesewedge))
+		to_chat(src, "<span class='green'>You eat [src], restoring some health.</span>")
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm]:
```suggestion
		to_chat(src, span_green("You eat [src], restoring some health."))
```

File: code/modules/mob/living/simple_animal/hostile/rat.dm [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+/mob/living/simple_animal/hostile/rat/handle_automated_action()
+	. = ..()
+	if (!mind)
+		if(prob(40))
+			var/turf/open/floor/F = get_turf(src)
+			if(istype(F) && !F.intact)
+				var/obj/structure/cable/C = locate() in F
+				if(C && prob(15))
+					if(C.avail())
+						visible_message("<span class='warning'>[src] chews through the [C]. It's toast!</span>")
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/hostile/rat.dm]:
```suggestion
						visible_message(span_warning("[src] chews through the [C]. It's toast!"))
```

File: code/modules/mob/living/simple_animal/hostile/rat.dm [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+						playsound(src, 'sound/effects/sparks2.ogg', 100, TRUE)
+						C.deconstruct()
+						death()
+				else if(C && C.avail())
+					visible_message("<span class='warning'>[src] chews through the [C]. It looks unharmed!</span>")
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/hostile/rat.dm]:
```suggestion
					visible_message(span_warning("[src] chews through the [C]. It looks unharmed!"))
```

File: code/modules/mob/living/simple_animal/hostile/rat.dm [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+		if(ratself.faction_check_mob(src, TRUE))
+			. += "<span class='notice'>You both serve the same king.</span>"
+		else
+			. += "<span class='warning'>This fool serves a different king!</span>"
+	else if(istype(user,/mob/living/simple_animal/hostile/regalrat))
+		var/mob/living/simple_animal/hostile/regalrat/ratking = user
+		if(ratking.faction_check_mob(src, TRUE))
+			. += "<span class='notice'>This rat serves under you.</span>"
+		else
+			. += "<span class='warning'>This peasant serves a different king! Strike him down!</span>"
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/hostile/rat.dm]:
```suggestion
/mob/living/simple_animal/hostile/rat/examine(mob/user)
	. = ..()
	if(istype(user,/mob/living/simple_animal/hostile/rat))
		var/mob/living/simple_animal/hostile/rat/ratself = user
		if(ratself.faction_check_mob(src, TRUE))
			. += span_notice("You both serve the same king.")
		else
			. += span_warning("This fool serves a different king!")
	else if(istype(user,/mob/living/simple_animal/hostile/regalrat))
		var/mob/living/simple_animal/hostile/regalrat/ratking = user
		if(ratking.faction_check_mob(src, TRUE))
			. += span_notice("This rat serves under you.")
		else
			. += span_warning("This peasant serves a different king! Strike him down!")
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 1 above)
Comment (ghost) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm:202]: For me it looks strange if separated

**Comment by bruhlookatthisdood** [comment]
https://www.youtube.com/watch?v=OXQwx1EolD8

**Comment by JamieD1** [comment]
Rats should eat cables, they literally are worse then mice.

**Comment by JamieD1** [comment]
> Rats should eat cables, they literally are worse then mice.

Scratch that, make them eat lots more cable more often.

**Comment by ghost** [comment]
They eat cables, but actually they will in most situations die after it, so making them eat cables more will be kinda useless

File: code/modules/mob/living/simple_animal/hostile/rat.dm [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+	if(is_type_in_list(A, GLOB.mouse_comestible))
+		return TRUE
+	if(istype(A, /obj/item/reagent_containers/food) && !(locate(/obj/structure/table) in get_turf(A)))
+		return TRUE
+
+
+
+
+
+
```
Comment (JamieD1) [inline code/modules/mob/living/simple_animal/hostile/rat.dm]:
```suggestion
```

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 7]
```diff
@@ -301,26 +218,18 @@
 		if (do_mob(src, target, 2 SECONDS))
 			target.reagents.add_reagent(/datum/reagent/rat_spit,rand(1,3),no_react = TRUE)
 			to_chat(src, span_notice("You finish licking [target]."))
+	else if(istype(target, /obj/item/reagent_containers/food/snacks/cheesewedge))
+		to_chat(src, span_green("You eat [src], restoring some health."))
+		heal_bodypart_damage(30)
+		qdel(target)
+
+
```
Comment (JamieD1) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm]:
```suggestion
```

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**